benchmark/reformat_input.py

```diff
@@ -337,19 +339,192 @@ def reformat_treepuzzle(gene_tree,
+        nucl_seq += str(gb)
+        for feature in gb.interval_metadata.features:
+            if feature['type_'] == 'CDS' and 'protein_id' in feature:
+                protein_id = feature['protein_id'].replace('\"', '')
+                if protein_id not in genes:
+                    translation = feature['translation'].replace(' ', '') \
+                        .replace('\"', '')
+                    strand = '-' if feature['rc_'] else '+'
+                    loc = gb.interval_metadata.features[feature]
+                    start = loc[0][0] + abs_pos + 1
+                    end = loc[0][1] + abs_pos
+                    genes[protein_id] = [translation, start, end, strand]
+        abs_pos += int(size)
+    gb = DNA(nucl_seq)
+    gb.metadata['LOCUS'] = {'locus_name': 'locus001', 'size': len(nucl_seq),
+                            'unit': 'bp', 'shape': 'circular',
+                            'division': 'CON', 'mol_type': 'DNA',
+                            'date': '01-JAN-1900'}
+    gb.metadata['id'] = 'locus001'
+    return (gb, genes)
+
+
+def _write_genebank_file(gb,
+                         genes,
+                         output_dir):
+    """ Write output genome to a GenBank file.
+
+    Parameters
+    ----------
+    gb: skbio.Sequence,
+        Genome sequence and metadata
+    genes: dict of { list of [ string, int, int, string ] }
+        Gene name : translation, start, end, and strand
+    output_dir: string
+        output directory path
+
+    Notes
+    -----
+    The reason this script does not use skbio's GenBank write function, is
+    that EGID cannot parse skbio-written GenBank format. Perhaps the reason
+    is the differential pattern of leading white spaces in the feature table.
+    """
+    output_f = open(join(output_dir, 'id.gbk'), 'w')
+    tmp_fp = join(output_dir, 'id.tmp')
+    DNA.write(gb, tmp_fp, format='genbank')
+    with open(tmp_fp, 'r') as input_f:
+        output_f.write(input_f.readline())
+    output_f.write('FEATURES             Location/Qualifiers\n')
+    gid = 1
+    for (gene, l) in sorted(genes.items(), key=lambda x: x[1][1]):
+        location = str(l[1]) + '..' + str(l[2])
+        if l[3] == '-':
+            location = 'complement(' + location + ')'
+        lines = []
+        lines.append('     gene            ' + location)
+        lines.append('                     /locus_tag=gene' + str(gid))
+        lines.append('     CDS             ' + location)
+        lines.append('                     /locus_tag=gene' + str(gid))
+        lines.append('                     /protein_id=' + gene)
+        lines.append('                     /translation=' + l[0])
+        for line in lines:
+            output_f.write(line + '\n')
+        gid += 1
+    with open(tmp_fp, 'r') as input_f:
+        next(input_f)
+        for line in input_f:
+            output_f.write(line)
+    output_f.close()
+    remove(tmp_fp)
+
+
+def reformat_egid(genbank_fp,
```

Comment:
I think this method suffers from the same errors than the previous ones, could you also review this and the following methods?